controller/iteration.go

```diff
@@ -232,6 +232,92 @@ func (c *IterationController) Update(ctx *app.UpdateIterationContext) error {
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrUnauthorized(err.Error()))
+	}
+	err = application.Transactional(c.db, func(appl application.Application) error {
+		itr, err := appl.Iterations().Load(ctx.Context, ctx.IterationID)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, err)
+		}
+		s, err := appl.Spaces().Load(ctx, itr.SpaceID)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, goa.ErrNotFound(err.Error()))
+		}
+		if !uuid.Equal(*currentUser, s.OwnerId) {
+			log.Warn(ctx, map[string]interface{}{
+				"space_id":     s.ID,
+				"space_owner":  s.OwnerId,
+				"current_user": *currentUser,
+			}, "user is not the space owner")
+			return jsonapi.JSONErrorResponse(ctx, errors.NewForbiddenError("user is not the space owner"))
+		}
+		if itr.IsRoot(s.ID) {
+			log.Warn(ctx, map[string]interface{}{
+				"space_id":     s.ID,
+				"iteration_id": itr.ID,
+			}, "can not delete root iteration")
+			return jsonapi.JSONErrorResponse(ctx, errors.NewForbiddenError("can not delete root iteration"))
+		}
+		subtree, err := appl.Iterations().LoadChildren(ctx, ctx.IterationID)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, err)
+		}
+		// Fetch parent iteration to which work items will get attached
+		parentID := itr.Parent()
+		if parentID == uuid.Nil {
+			return jsonapi.JSONErrorResponse(ctx, goa.ErrNotFound("can not find parent iteration"))
+		}
+		parentIteration, err := appl.Iterations().Load(ctx, parentID)
+		if err != nil {
+			log.Error(ctx, map[string]interface{}{
+				"iteration_id": parentID,
+				"err":          err.Error(),
+			}, "unable to load iteration")
+			return jsonapi.JSONErrorResponse(ctx, err)
+		}
+		// delete all children along with given iteration
+		subtree = append(subtree, *itr)
+		for _, child := range subtree {
+			// fetch associated work items
+			wis, err := appl.WorkItems().LoadByIteration(ctx, child.ID)
+			if err != nil {
+				return jsonapi.JSONErrorResponse(ctx, err)
+			}
+			// update iteration on all associated work items
+			for _, wi := range wis {
+				// move WI to root iteration
+				wi.Fields[workitem.SystemIteration] = parentIteration.ID.String()
+				_, err = appl.WorkItems().Save(ctx, wi.SpaceID, *wi, *currentUser)
+				if err != nil {
+					log.Error(ctx, map[string]interface{}{
+						"workitem_id": wi.ID,
+						"err":         err.Error(),
+					}, "unable to update iteration for work item")
+					return jsonapi.JSONErrorResponse(ctx, err)
+				}
+			}
+			// now, remove the iteration
+			err = appl.Iterations().Delete(ctx.Context, child.ID)
+			if err != nil {
+				log.Error(ctx, map[string]interface{}{
+					"iteration_id": child.ID,
+					"err":          err.Error(),
+				}, "unable to delete iteration")
+				return jsonapi.JSONErrorResponse(ctx, err)
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, errors.NewInternalError(ctx, err))
+	}
+	return ctx.OK([]byte{})
```

Comment:
I know we are using `200 OK` response everywhere else with empty body in the response. But this not correct as per [RFC 2616](https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html#sec9.7). It recommends to use `204`, if no content.

Reply:
Here is an example from GitHub REST API: https://developer.github.com/v3/gists/comments/#delete-a-comment
(They are using `204`)

Reply:
In Goa, you can do it like this: `return ctx.NoContent()`

Reply:
yes @baijum thanks, good point!
changed in https://github.com/fabric8-services/fabric8-wit/pull/1683/commits/f838edb05160aee4f9680a110e02a5e5e1709950